### PR TITLE
[flutter_adaptive_scaffold] Fix breakpoint not being active in certain cases like foldables

### PR DESCRIPTION
**Pull request by martijn00**
*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

As pointed out by the docs on this:

> Available width and height are classified separately, so at any point in time, your app has two window size classes—one for width, one for height. Available width is usually more important than available height due to the ubiquity of vertical scrolling, so the width window size class is likely more relevant to your app's UI.

*List which issues are fixed by this PR. You must list at least one issue.*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

**Comment by hanskokx** [comment]
@martijn00 Tested and confirmed working on a Pixel Fold emulator using the following pubspec.yaml settings:

```yaml
  flutter_adaptive_scaffold:
    git:
      url: https://github.com/martijn00/flutter_packages.git
      ref: af02e0e05adda36f22755cafd27575602ad95030
      path: packages/flutter_adaptive_scaffold
```

**Comment by martijn00** [comment]
@Renzo-Olivares can you check this one?

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: packages/flutter_adaptive_scaffold/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.2.4
+
+* Fix breakpoint not being active in certain cases like foldables
```
Comment (Renzo-Olivares) [inline packages/flutter_adaptive_scaffold/CHANGELOG.md:3]: nit: add period at the end of sentence.

File: packages/flutter_adaptive_scaffold/lib/src/breakpoints.dart [hunk 2]
```diff
@@ -294,11 +294,10 @@ class Breakpoint {
         ? width >= lowerBoundWidth
         : width >= lowerBoundWidth && width < upperBoundWidth;
 
-    final bool isHeightActive = isDesktop ||
-        orientation == Orientation.portrait ||
-        (orientation == Orientation.landscape && andUp
+    final bool isHeightActive =
```
Comment (Renzo-Olivares) [inline packages/flutter_adaptive_scaffold/lib/src/breakpoints.dart:297]: Can this just be `isPortrait || isWidthActive || andUp`?

Comment (martijn00) [reply packages/flutter_adaptive_scaffold/lib/src/breakpoints.dart:297]: I'm in doubt how to properly handle all cases. According to the specs it should check height, but it is hard to handle edge cases. The new tests cover a lot of it.

Comment (Renzo-Olivares) [reply packages/flutter_adaptive_scaffold/lib/src/breakpoints.dart:297]: Sorry, I didn't mean the entire case should be only that, just the first line. Because if `isPortrait` is false then the first case will fail and check the second case, I don't think we have to check if `isPortrait` is false in the second case. Or would this fail?

Comment (martijn00) [reply packages/flutter_adaptive_scaffold/lib/src/breakpoints.dart]: Aah right. Fixed now.

**Comment by martijn00** [comment]
@hanskokx this is now released. Can you confirm everything is good again?

**Comment by hanskokx** [comment]
> @hanskokx this is now released. Can you confirm everything is good again?

Already saw and tested it. Everything looks good and works for me! Thanks for the quick fix :)
